staging/src/k8s.io/kubectl/pkg/cmd/portforward/portforward.go

```diff
@@ -225,6 +226,70 @@ func convertPodNamedPortToNumber(ports []string, pod corev1.Pod) ([]string, erro
 	return converted, nil
 }
 
+// checkUDPPortInService returns an error if remote port in Service is a UDP port
+// TODO: remove this check after #47862 is resolved
+func checkUDPPortInService(ports []string, svc *corev1.Service) error {
+	udpOnlyPorts := sets.NewInt()
+	for _, port := range svc.Spec.Ports {
+		portNum := int(port.Port)
+		switch port.Protocol {
+		case corev1.ProtocolUDP:
+			udpOnlyPorts.Insert(portNum)
+		case corev1.ProtocolTCP:
+			udpOnlyPorts.Delete(portNum)
+		default:
+		}
+	}
+	for _, port := range ports {
```

Comment:
Extract the logic from this for loop to a new method, since it's identical with that one from `checkUDPPortInPod`

Reply:
hmm they are slightly different. 
In `checkUDPPortInService`, `util.LookupServicePortNumberByName` is invoked to lookup port by name, while in `checkUDPPortInPod`, `util.LookupContainerPortNumberByName` is called.

Reply:
@soltysh I have extracted the logic to a new method `checkUDPPorts`, it will check the type of object and call corresponding function to lookup port by name.